File: sdk/r/R/cloud_gpu.R
#' Cloud GPU: GPU instance management
#' @param http HttpClient object
#' @return Resource object
#' @export
CloudGpuResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "CloudGpuResource"
  env
}

#' PUT /api/service/{id}/vms/reboot - Reboot virtual server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
reboot <- function(self, ...) {
  self$http$put("/api/service/{id}/vms/reboot", list(...))
}

#' PUT /api/service/{id}/vms/stop - Stop virtual server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
stop <- function(self, ...) {
  self$http$put("/api/service/{id}/vms/stop", list(...))
}

#' PUT /api/service/{id}/vms/start - Start virtual server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
start <- function(self, ...) {
  self$http$put("/api/service/{id}/vms/start", list(...))
}

#' GET /api/service/{id}/vms/firewall - Get firewall rules
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getFirewallRules <- function(self, ...) {
  self$http$get("/api/service/{id}/vms/firewall", list(...))
}

#' POST /api/service/{id}/vms/firewall - Add firewall rules
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
addFirewallRules <- function(self, ...) {
  self$http$post("/api/service/{id}/vms/firewall", list(...))
}

#' DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
removeFirewallRule <- function(self, ...) {
  self$http$delete("/api/service/{id}/vms/firewall/{position}", list(...))
}

File: sdk/c/src/cloud_gpu.c
#include "hitechcloud/cloud_gpu.h"
#include <stdlib.h>


/* PUT /api/service/{id}/vms/reboot - Reboot virtual server */
char *hc_cloud_gpu_reboot(hc_http_client_t *client) {
    return hc_put(client, "/api/service/{id}/vms/reboot", NULL);
}

/* PUT /api/service/{id}/vms/stop - Stop virtual server */
char *hc_cloud_gpu_stop(hc_http_client_t *client) {
    return hc_put(client, "/api/service/{id}/vms/stop", NULL);
}

/* PUT /api/service/{id}/vms/start - Start virtual server */
char *hc_cloud_gpu_start(hc_http_client_t *client) {
    return hc_put(client, "/api/service/{id}/vms/start", NULL);
}

/* GET /api/service/{id}/vms/firewall - Get firewall rules */
char *hc_cloud_gpu_get_firewall_rules(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/vms/firewall", NULL);
}

/* POST /api/service/{id}/vms/firewall - Add firewall rules */
char *hc_cloud_gpu_add_firewall_rules(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/vms/firewall", NULL);
}

/* DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule */
char *hc_cloud_gpu_remove_firewall_rule(hc_http_client_t *client) {
    return hc_delete(client, "/api/service/{id}/vms/firewall/{position}", NULL);
}

